packages/base-manager/src/manager-base.ts

```diff
@@ -330,35 +342,57 @@ export abstract class ManagerBase implements IWidgetManager {
     serialized_state: any = {}
   ): Promise<WidgetModel> {
     const model_id = options.model_id;
-    const model_promise = this.loadClass(
+    const model_promise = this.loadModelClass(
       options.model_name,
       options.model_module,
       options.model_module_version
-    ) as Promise<typeof WidgetModel>;
+    );
     let ModelType: typeof WidgetModel;
+
+    const makeErrorModel = (error: any, msg: string) => {
+      const Cls = widgets.createErrorWidget(error, msg);
+      const widget_model = new Cls({
+        _model_module: options.model_module,
+        _model_name: options.model_name,
```

Comment:
Do we need to pass those options? IIUC those are for creating the actual widget, not the error widget.